base_layer/wallet_ffi/src/lib.rs

```diff
@@ -9215,22 +9052,8 @@ pub unsafe extern "C" fn wallet_destroy(wallet: *mut TariWallet) {
     if !wallet.is_null() {
         debug!(target: LOG_TARGET, "Wallet pointer not yet destroyed, shutting down now");
         let mut w = Box::from_raw(wallet);
-        let wallet_comms = w.wallet.comms.clone();
         w.shutdown.trigger();
         w.runtime.block_on(w.wallet.wait_until_shutdown());
-        // The wallet should be shutdown by now; these are just additional confirmations
-        loop {
-            if w.shutdown.is_triggered() &&
```

Comment:
This code makes no sense  and doesnt solve a bug "where the comms took a bit long to shutdown"

The line above will wait until `wait_until_shutdown()` has resolved - at which point it is shut down. So if it takes long to shut down, then it would wait in the above line. 

I'd suggest rather timing out if this takes long - however I don't think this is an issue.

```rust
    loop {
            if w.shutdown.is_triggered() &&
                wallet_comms.shutdown_signal().is_triggered() &&
                w.runtime
                    .block_on(wallet_comms.connectivity().get_connectivity_status())
                    .is_err()
            {
                break;
            };
            w.runtime
                .block_on(async { tokio::time::sleep(Duration::from_millis(250)).await });
        
```